### PR TITLE
TOTK: optimize item selection feature

### PR DESCRIPTION
**Pull request by xiyuesaves**
Hello, I tried to optimize the item selection function and added the ability to search for items, which will make it more convenient to use😀
file

### CODE REVIEW

**Comment by marcrobledo** [comment]
Haven't had a chance to try it but looks quite good!

Only downside I can see is that the ability to browse with up/down keyboard arrows is probably lost.
Also, image lazy loading would be great (unless it's already implemented!).

**Comment by xiyuesaves** [comment]
Sorry, I have no way to solve the problem of not being able to use the up and down keys to switch between selections 😢, adding the 'lazy' attribute to the image element should be automatically handled by the browser

**Comment by marcrobledo** [comment]
I'll merge this one for sure, I promise. But give me some time please, got swamped with other things.

I'll also implement the keyboard selection.

**Comment by xiyuesaves** [comment]
I completely understand. Life always brings a lot of things to handle, and sometimes it can truly feel overwhelming. Please try to focus your attention on what you're currently dealing with.
Also, I've done the keyboard selection function and it looks pretty good

[video.webm](https://github.com/marcrobledo/savegame-editors/assets/25543545/25554651-312b-4dcb-8b04-f66700a9e85b)

**Comment by marcrobledo** [comment]
Working on it!

I'm gonna add a few of little QoL changes if you don't mind.

File: zelda-totk/zelda-totk.class.pouch.js [hunk 1]
```diff
@@ -202,6 +202,26 @@ Pouch.updateItemIcon=function(item){
 	else
 		item._htmlIcon.src=ICON_PATH+item.category+'/'+item.id+'.png';
 };
+Pouch.getItemIcon = function (item) {
```
Comment (marcrobledo) [inline zelda-totk/zelda-totk.class.pouch.js:205]: This is a good id!
Although, it is repeating code from updateItemIcon, will remove code from the former.

File: zelda-totk/zelda-totk.css [hunk 2]
```diff
@@ -929,3 +929,133 @@ label.clickable:has(> input[type=radio]:checked){
...
+  transition: 300ms;
+}
+.editItem .search-input::placeholder{
+  font-size: 15px;
+  color: rgba(255, 255, 255, 0.8);
+}
+.editItem .search-input:hover{
+  border: solid 1px #666666;
+}
+.edit-item-mask{
```
Comment (marcrobledo) [inline zelda-totk/zelda-totk.css:1047]: Gonna try to get rid of the `edit-item-mask` element, I'm pretty use it's not necessary if we bind the blur event correctly to the search filter input.

File: zelda-totk/zelda-totk.js [hunk 3]
```diff
@@ -514,41 +515,89 @@ SavegameEditor={
...
-		if(this.selectItem.lastCategory !== item.category){
-			this.selectItem.innerHTML='';
-			var itemList=this.getAvailableItems(item.category);
-			for(var i=0; i<itemList.length; i++){
-				var opt=document.createElement('option');
-				opt.value=itemList[i];
-				opt.innerHTML=_(itemList[i]);
-				this.selectItem.appendChild(opt);
-			}
+    document.querySelector("#header").classList.add("disable-pointer");
```
Comment (marcrobledo) [inline zelda-totk/zelda-totk.js:519]: I'm going to switch to jQuery for some of these, just for consistency purposes.

File: zelda-totk/zelda-totk.css (hunk 2 above)
Comment (xiyuesaves) [reply zelda-totk/zelda-totk.css:1047]: The reason for using this is that if the option list is clicked by the mouse, it triggers the blur event of the input first, removing the option list before the option is clicked, leading to a failed selection. Perhaps a custom event could be used to achieve this function.

**Comment by marcrobledo** [comment]
I finally managed to finish this!

Let me know if you have any questions on all changes I did, I'll be glad to answer.

**Pull request merged**